lib/base_service.ts

```diff
@@ -51,11 +53,20 @@ export interface Credentials {
   username: string;
   password: string;
   api_key: string;
+  platform_api_key?: string;
   url: string;
 }
 
+export interface TokenData {
+  access_token?: string;
+  refresh_token?: string;
+  token_type?: string;
+  expires_in?: number;
+  expiration?: number;
+}
```

Comment:
Might be worth renaming this to `IAMTokenData`, since other token providers could have different fields.

(For example, [Watson Tokens][1] only return the `access_token` last I checked. There's no information in the response about a `refresh_token` or expiration.)

[1]: https://console.bluemix.net/docs/services/watson/getting-started-tokens.html

Reply:
Would it also be reasonable to make some or all of these fields `required` instead of `optional`?

Reply:
> Might be worth renaming this to IAMTokenData, since other token providers could have different fields.
I think that's a good idea!

> Would it also be reasonable to make some or all of these fields required instead of optional?
I was using optional because of the method we decided on to determine if the API supported tokens from the Swagger, which required the object to be set to `{}`. I am not doing this anymore, so it may make sense for the properties to be required.